packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1819,40 +1828,22 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         if (this.canSendOps()) {
             if (this.context.submitBatchFn !== undefined) {
                 const batchToSend: IBatchMessage[] = [];
+
+                let batchContentsLength = 0;
                 for (const message of batch) {
-                    let contents = message.contents;
-                    let metadata = message.metadata;
-                    if (this.runtimeOptions.compressionOptions.minimumSize &&
-                        this.runtimeOptions.compressionOptions.minimumSize < message.contents.length) {
-                        this.compressedOpCount++;
-                        const copiedMessage = { ...message.deserializedContent };
-
-                        const compressionStart = Date.now();
-                        const contentsAsBuffer = new TextEncoder().encode(JSON.stringify(copiedMessage.contents));
-                        const compressedContents = compress(contentsAsBuffer);
-                        const compressedContent = IsoBuffer.from(compressedContents).toString("base64");
-                        const duration = Date.now() - compressionStart;
-
-                        if (this.compressedOpCount % 100) {
-                            this.mc.logger.sendPerformanceEvent({
-                                eventName: "compressedOp",
-                                duration,
-                                sizeBeforeCompression: message.contents.length,
-                                sizeAfterCompression: compressedContent.length,
-                            });
-                        }
-
-                        copiedMessage.contents = compressedContent;
-                        const stringifiedContents = JSON.stringify(copiedMessage);
-
-                        if (stringifiedContents.length < message.contents.length) {
-                            contents = JSON.stringify(copiedMessage);
-                            metadata = { ...message.metadata, compressed: true };
-                        }
-                    }
+                    batchContentsLength += message.contents.length;
+                }
 
-                    batchToSend.push({ contents, metadata });
+                if (this.runtimeOptions.compressionOptions?.minimumSize !== undefined
+                    && this.runtimeOptions.compressionOptions.compressionAlgorithm !== undefined
+                    && batchContentsLength > this.runtimeOptions.compressionOptions.minimumSize) {
+                        batchToSend.push(...this.opCompressor.compressBatch(batch, batchContentsLength));
```

Comment:
Without looking at actual implementation, I feel API to be a bit weird RE compression flow. It feels like it would be easier to deal with it and make more sense from API perspective if OpCompressor instance was created for just one batch, and things like batchContentsLength were passed in constructor. Also see above comment - it's likely even easier of it's done as part of BatchManager flow itself (i.e. BatchManager.popBatch() would do this transformation)

Reply:
Oh, sorry, I thought we were pushing ops one by one into it. This is fine as is, but please consider pushing invocation into BatchManager.

Reply:
Moved into the BatchManager